packages/block-library/src/post-date/edit.js

```diff
@@ -35,30 +35,39 @@ export default function PostDateEdit( {
 } ) {
 	const isDescendentOfQueryLoop = Number.isFinite( queryId );
 	const [ siteFormat ] = useEntityProp( 'root', 'site', 'date_format' );
+	const [ timeFormat ] = useEntityProp( 'root', 'site', 'time_format' );
 	const [ date, setDate ] = useEntityProp(
 		'postType',
 		postType,
 		'date',
 		postId
 	);
-	const settings = __experimentalGetSettings();
-	// To know if the current time format is a 12 hour time, look for "a".
-	// Also make sure this "a" is not escaped by a "/".
-	const is12Hour = /a(?!\\)/i.test(
-		settings.formats.time
-			.toLowerCase() // Test only for the lower case "a".
-			.replace( /\\\\/g, '' ) // Replace "//" with empty strings.
-			.split( '' )
-			.reverse()
-			.join( '' ) // Reverse the string and test for "a" not followed by a slash.
-	);
-	const formatOptions = Object.values( settings.formats ).map(
-		( formatOption ) => ( {
-			key: formatOption,
-			name: dateI18n( formatOption, date ),
-		} )
-	);
-	const resolvedFormat = format || siteFormat || settings.formats.date;
+	// To know if the time format is a 12 hour time, look for any of the 12 hour
+	// format characters: 'a', 'A', 'g', and 'h'. The character must be
+	// unescaped, i.e. not preceded by a '\'. Coincidentally, 'aAgh' is how I
+	// feel when working with regular expressions.
+	// https://www.php.net/manual/en/datetime.format.php
+	const is12Hour = /(?:^|[^\\])[aAgh]/.test( timeFormat );
+	const formatOptions = [
+		{
+			key: null,
+			name: sprintf(
+				// translators: %s: Example of how the date will look if selected.
+				_x( 'Site default (%s)', 'date format option' ),
+				dateI18n( siteFormat, date )
+			),
+		},
+		...[
+			_x( 'M jS', 'date format option' ),
+			_x( 'M j, Y', 'date format option' ),
+			_x( 'M j, Y, h:i A', 'date format option' ),
+			_x( 'F j, Y', 'date format option' ),
+			_x( 'd/m/Y', 'date format option' ),
+		].map( ( suggestedFormat ) => ( {
+			key: suggestedFormat,
+			name: dateI18n( suggestedFormat, date ),
+		} ) ),
```

Comment:
Probably an issue in trunk too, so it might be something to fix separately, but thought I'd mention it as it might be a fix we can roll in to this PR.

The labels could be improved for screenreaders. Both 'Feb 25, 2022' and 'Februrary 25, 2022' are announced as 'February 25th 2022'. I guess Voiceover does some normalisation of dates. I'm not sure if we can add some screenreader text into the `CustomSelectControl` items that also contains the format.

Let me know if you want me to create a separate issue.

Reply:
Oh interesting. That's tricky. I'm not entirely sure what the expectations would be for folks using screen readers. Go ahead and create a new issue so that we can ask for feedback on that from a11y folks.

Reply:
Here's a new issue - https://github.com/WordPress/gutenberg/issues/39278

Testing again, and there's also no way to tell which item in the dropdown is currently checked. 😞

Reply:
I just had a quick look at the SelectControl component, and it looks like it doesn't currently set the `selected` attribute on option elements ([here](https://github.com/wordpress/gutenberg/blob/d5915916abc45e6682f4bdb70888aa41e98aa395/packages/components/src/select-control/index.tsx#L152)). I was wondering if that could be the culprit? (Probably something to look at separately to this PR?)